### DISCUSSION TITLE
using SSH to connect to the git repository

### DISCUSSION THREAD

**Discussion by pettyla** [gocd/gocd#9757] [Q&A]
Whether to support using SSH to connect to the git repository. Or Is there a plug-in that solves this scenario?

**Comment by chadlwilson** [comment]
You can use GoCD to connect with SSH to git repos, but it doesn't have any support for helping you manage these SSH keys on the host with which GoCD is running, so you'll have to manage the contents of `~/.ssh/config` yourself and ensure the GoCD server and agent processes have access to relevant private keys for the various repos. If using Docker or Kubernetes this would probably mean mounting these into the container in the right place somehow; and possibly using standard `GIT_SSH` env vars to [control how Git does SSH](https://git-scm.com/book/en/v2/Git-Internals-Environment-Variables).

There are a few similar threads on the Google Group where you might be able to find some other info, e.g https://groups.google.com/g/go-cd/c/ZRJYq_W-8-I/m/gQxgPUyJBAAJ
